### ISSUE TITLE
Error: internal compiler error: TyKind::Error constructed but no error reported

### ISSUE THREAD

**Issue by thomasyonug** [rust-lang/rust#114984]
I tried this code:

```rust
#![feature(inherent_associated_types)]
#![allow(incomplete_features)]

struct Foo<T>(T);

impl<'a> Foo<fn(&'a ())> {
    type Assoc = &'a ();

    fn foo(&self, f: Self::Assoc) {
        let mut i = 0;
        for _ in 0..10 {
            i = i + 1;
        }
        println!("i: {}", i);
    }
}

fn main() {}
```

I expected to see this happen: compiled

Instead, this happened: 
```bash
error: internal compiler error: no errors encountered even though `delay_span_bug` issued

error: internal compiler error[E0391]: cycle detected when computing function signature of `<impl at a.rs:6:1: 6:25>::foo`
  --> a.rs:9:5
   |
9  |     fn foo(&self, f: Self::Assoc) {
```

### Meta

`rustc --version --verbose`:
```
rustc 1.73.0-nightly (439d066bc 2023-08-10)
```

<details><summary>Backtrace</summary>
<p>

```
delayed span bug: 
   0: <rustc_errors::HandlerInner>::emit_diagnostic
   1: <rustc_errors::Handler>::emit_diagnostic
   2: <rustc_span::ErrorGuaranteed as rustc_errors::diagnostic_builder::EmissionGuarantee>::diagnostic_builder_emit_producing_guarantee
   3: rustc_query_system::query::plumbing::mk_cycle::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt>
   4: rustc_query_system::query::plumbing::cycle_error::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt>
   5: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
   6: rustc_query_impl::query_impl::fn_sig::get_query_non_incr::__rust_end_short_backtrace
   7: rustc_middle::query::plumbing::query_get_at::<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>>
   8: <rustc_hir_analysis::variance::constraints::ConstraintContext>::build_constraints_for_item
   9: rustc_hir_analysis::variance::crate_variances
  10: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::crate_variances::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
  11: <rustc_query_impl::query_impl::crate_variances::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
  12: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  13: rustc_query_impl::query_impl::crate_variances::get_query_non_incr::__rust_end_short_backtrace
  14: rustc_hir_analysis::variance::variances_of
  15: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::variances_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 16]>>
  16: <rustc_query_impl::query_impl::variances_of::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
  17: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 16]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  18: rustc_query_impl::query_impl::variances_of::get_query_non_incr::__rust_end_short_backtrace
  19: rustc_middle::ty::relate::structurally_relate_tys::<rustc_infer::infer::sub::Sub>
  20: <rustc_infer::infer::sub::Sub as rustc_middle::ty::relate::TypeRelation>::tys
  21: <rustc_infer::infer::InferCtxt>::commit_if_ok::<rustc_infer::infer::InferOk<()>, rustc_middle::ty::error::TypeError, <rustc_infer::infer::at::Trace>::sub<rustc_middle::ty::Ty>::{closure#0}>
  22: <rustc_trait_selection::traits::engine::ObligationCtxt>::sup::<rustc_middle::ty::Ty>
  23: <rustc_infer::infer::InferCtxt>::probe::<bool, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}::{closure#0}>
  24: <&mut <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0} as core::ops::function::FnMut<(&(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)),)>>::call_mut
  25: <core::iter::adapters::copied::Copied<core::slice::iter::Iter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>> as core::iter::traits::iterator::Iterator>::try_fold::<(), core::iter::traits::iterator::Iterator::find::check<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)), &mut <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}>::{closure#0}, core::ops::control_flow::ControlFlow<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>>
  26: <alloc::vec::Vec<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))> as alloc::vec::spec_from_iter::SpecFromIter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)), core::iter::adapters::filter::Filter<core::iter::adapters::copied::Copied<core::slice::iter::Iter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>>, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}>>>::from_iter
  27: <rustc_infer::infer::InferCtxt>::probe::<alloc::vec::Vec<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}>
  28: <dyn rustc_hir_analysis::astconv::AstConv>::associated_path_to_ty::{closure#0}
  29: <dyn rustc_hir_analysis::astconv::AstConv>::ast_ty_to_ty_inner::{closure#0}
  30: <dyn rustc_hir_analysis::astconv::AstConv>::ty_of_fn::{closure#0}
  31: rustc_hir_analysis::collect::infer_return_ty_for_fn_sig
  32: rustc_hir_analysis::collect::fn_sig
  33: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::fn_sig::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 24]>>
  34: <rustc_query_impl::query_impl::fn_sig::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
  35: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  36: rustc_query_impl::query_impl::fn_sig::get_query_non_incr::__rust_end_short_backtrace
  37: <rustc_hir_analysis::collect::CollectItemTypesVisitor as rustc_hir::intravisit::Visitor>::visit_impl_item
  38: <rustc_middle::hir::map::Map>::visit_item_likes_in_module::<rustc_hir_analysis::collect::CollectItemTypesVisitor>
  39: rustc_hir_analysis::collect::collect_mod_item_types
  40: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::collect_mod_item_types::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 0]>>
  41: <rustc_query_impl::query_impl::collect_mod_item_types::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId)>>::call_once
  42: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::Erased<[u8; 0]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  43: rustc_query_impl::query_impl::collect_mod_item_types::get_query_non_incr::__rust_end_short_backtrace
  44: <rustc_middle::hir::map::Map>::for_each_module::<rustc_hir_analysis::check_crate::{closure#0}::{closure#0}::{closure#0}>
  45: rustc_hir_analysis::check_crate
  46: rustc_interface::passes::analysis
  47: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 1]>>
  48: <rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
  49: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 1]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  50: rustc_query_impl::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  51: <rustc_middle::ty::context::GlobalCtxt>::enter::<rustc_driver_impl::run_compiler::{closure#1}::{closure#2}::{closure#6}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
  52: rustc_span::set_source_map::<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  53: std::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
  54: <<std::thread::Builder>::spawn_unchecked_<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  55: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/alloc/src/boxed.rs:2007:9
  56: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/alloc/src/boxed.rs:2007:9
  57: std::sys::unix::thread::Thread::new::thread_start
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/std/src/sys/unix/thread.rs:108:17
  58: start_thread
             at /build/glibc-S9d2JN/glibc-2.27/nptl/pthread_create.c:463
  59: __GI___clone
             at /build/glibc-S9d2JN/glibc-2.27/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95

delayed span bug: TyKind::Error constructed but no error reported
   0: <rustc_errors::HandlerInner>::emit_diagnostic
   1: <rustc_errors::Handler>::delay_span_bug::<rustc_span::span_encoding::Span, &str>
   2: <rustc_middle::ty::Ty>::new_misc_error
   3: <rustc_middle::ty::sty::Binder<rustc_middle::ty::sty::FnSig> as rustc_query_system::values::Value<rustc_middle::ty::context::TyCtxt, rustc_middle::dep_graph::dep_node::DepKind>>::from_cycle_error
   4: <rustc_query_impl::query_impl::fn_sig::dynamic_query::{closure#3} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, &[rustc_query_system::query::job::QueryInfo<rustc_middle::dep_graph::dep_node::DepKind>])>>::call_once
   5: rustc_query_system::query::plumbing::mk_cycle::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt>
   6: rustc_query_system::query::plumbing::cycle_error::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt>
   7: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
   8: rustc_query_impl::query_impl::fn_sig::get_query_non_incr::__rust_end_short_backtrace
   9: rustc_middle::query::plumbing::query_get_at::<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>>
  10: <rustc_hir_analysis::variance::constraints::ConstraintContext>::build_constraints_for_item
  11: rustc_hir_analysis::variance::crate_variances
  12: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::crate_variances::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 8]>>
  13: <rustc_query_impl::query_impl::crate_variances::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
  14: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 8]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  15: rustc_query_impl::query_impl::crate_variances::get_query_non_incr::__rust_end_short_backtrace
  16: rustc_hir_analysis::variance::variances_of
  17: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::variances_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 16]>>
  18: <rustc_query_impl::query_impl::variances_of::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
  19: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 16]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  20: rustc_query_impl::query_impl::variances_of::get_query_non_incr::__rust_end_short_backtrace
  21: rustc_middle::ty::relate::structurally_relate_tys::<rustc_infer::infer::sub::Sub>
  22: <rustc_infer::infer::sub::Sub as rustc_middle::ty::relate::TypeRelation>::tys
  23: <rustc_infer::infer::InferCtxt>::commit_if_ok::<rustc_infer::infer::InferOk<()>, rustc_middle::ty::error::TypeError, <rustc_infer::infer::at::Trace>::sub<rustc_middle::ty::Ty>::{closure#0}>
  24: <rustc_trait_selection::traits::engine::ObligationCtxt>::sup::<rustc_middle::ty::Ty>
  25: <rustc_infer::infer::InferCtxt>::probe::<bool, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}::{closure#0}>
  26: <&mut <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0} as core::ops::function::FnMut<(&(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)),)>>::call_mut
  27: <core::iter::adapters::copied::Copied<core::slice::iter::Iter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>> as core::iter::traits::iterator::Iterator>::try_fold::<(), core::iter::traits::iterator::Iterator::find::check<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)), &mut <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}>::{closure#0}, core::ops::control_flow::ControlFlow<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>>
  28: <alloc::vec::Vec<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))> as alloc::vec::spec_from_iter::SpecFromIter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId)), core::iter::adapters::filter::Filter<core::iter::adapters::copied::Copied<core::slice::iter::Iter<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>>, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}::{closure#0}>>>::from_iter
  29: <rustc_infer::infer::InferCtxt>::probe::<alloc::vec::Vec<(rustc_span::def_id::DefId, (rustc_span::def_id::DefId, rustc_span::def_id::DefId))>, <dyn rustc_hir_analysis::astconv::AstConv>::lookup_inherent_assoc_ty::{closure#1}>
  30: <dyn rustc_hir_analysis::astconv::AstConv>::associated_path_to_ty::{closure#0}
  31: <dyn rustc_hir_analysis::astconv::AstConv>::ast_ty_to_ty_inner::{closure#0}
  32: <dyn rustc_hir_analysis::astconv::AstConv>::ty_of_fn::{closure#0}
  33: rustc_hir_analysis::collect::infer_return_ty_for_fn_sig
  34: rustc_hir_analysis::collect::fn_sig
  35: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::fn_sig::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 24]>>
  36: <rustc_query_impl::query_impl::fn_sig::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::DefId)>>::call_once
  37: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::DefaultCache<rustc_span::def_id::DefId, rustc_middle::query::erase::Erased<[u8; 24]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  38: rustc_query_impl::query_impl::fn_sig::get_query_non_incr::__rust_end_short_backtrace
  39: <rustc_hir_analysis::collect::CollectItemTypesVisitor as rustc_hir::intravisit::Visitor>::visit_impl_item
  40: <rustc_middle::hir::map::Map>::visit_item_likes_in_module::<rustc_hir_analysis::collect::CollectItemTypesVisitor>
  41: rustc_hir_analysis::collect::collect_mod_item_types
  42: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::collect_mod_item_types::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 0]>>
  43: <rustc_query_impl::query_impl::collect_mod_item_types::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, rustc_span::def_id::LocalDefId)>>::call_once
  44: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::VecCache<rustc_span::def_id::LocalDefId, rustc_middle::query::erase::Erased<[u8; 0]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  45: rustc_query_impl::query_impl::collect_mod_item_types::get_query_non_incr::__rust_end_short_backtrace
  46: <rustc_middle::hir::map::Map>::for_each_module::<rustc_hir_analysis::check_crate::{closure#0}::{closure#0}::{closure#0}>
  47: rustc_hir_analysis::check_crate
  48: rustc_interface::passes::analysis
  49: rustc_query_impl::plumbing::__rust_begin_short_backtrace::<rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle::query::erase::Erased<[u8; 1]>>
  50: <rustc_query_impl::query_impl::analysis::dynamic_query::{closure#2} as core::ops::function::FnOnce<(rustc_middle::ty::context::TyCtxt, ())>>::call_once
  51: rustc_query_system::query::plumbing::try_execute_query::<rustc_query_impl::DynamicConfig<rustc_query_system::query::caches::SingleCache<rustc_middle::query::erase::Erased<[u8; 1]>>, false, false, false>, rustc_query_impl::plumbing::QueryCtxt, false>
  52: rustc_query_impl::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  53: <rustc_middle::ty::context::GlobalCtxt>::enter::<rustc_driver_impl::run_compiler::{closure#1}::{closure#2}::{closure#6}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
  54: rustc_span::set_source_map::<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  55: std::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
  56: <<std::thread::Builder>::spawn_unchecked_<rustc_interface::util::run_in_thread_pool_with_globals<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustc_driver_impl::run_compiler::{closure#1}>::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#0}::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>::{closure#1} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  57: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/alloc/src/boxed.rs:2007:9
  58: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/alloc/src/boxed.rs:2007:9
  59: std::sys::unix::thread::Thread::new::thread_start
             at /rustc/439d066bcf9496b1b8c8dde8bef3bce607a621bb/library/std/src/sys/unix/thread.rs:108:17
  60: start_thread
             at /build/glibc-S9d2JN/glibc-2.27/nptl/pthread_create.c:463
  61: __GI___clone
             at /build/glibc-S9d2JN/glibc-2.27/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95



rustc version: 1.73.0-nightly (439d066bc 2023-08-10)
platform: x86_64-unknown-linux-gnu

query stack during panic:
end of query stack

```

</p>
</details>

**Comment by fmease** [comment]
Duplicate of #110106.

@rustbot label -needs-triage I-ICE T-compiler F-inherent_associated_types

**Comment by compiler-errors** [comment]
Yeah, I'll close it as a dupe. Thanks @fmease.
